content/intro-to-storybook/react/en/screen.md

```diff
@@ -166,114 +183,166 @@ function App() {
+
+In your terminal, run the following command to generate a generic service worker inside your `public` folder:
+
+```shell
+yarn init-msw
+```
 
-## Supplying context with decorators
+Then, we'll need to update our `.storybook/preview.js` and initialize them:
 
-The good news is that it is easy to supply a Redux store to the `PureInboxScreen` in a story! We can just use a mocked version of the Redux store provided in a decorator:
+```diff:title=.storybook/preview.js
+import '../src/index.css';
 
-```diff:title=src/components/PureInboxScreen.stories.js
++ // Registers the msw addon
++ import { initialize, mswDecorator } from 'msw-storybook-addon';
+
++ // Initialize MSW
++ initialize();
+
++ // Provide the MSW addon decorator globally
++ export const decorators = [mswDecorator];
+
+//👇 Configures Storybook to log the actions( onArchiveTask and onPinTask ) in the UI.
+export const parameters = {
+  actions: { argTypesRegex: '^on[A-Z].*' },
+  controls: {
+    matchers: {
+      color: /(background|color)$/i,
+      date: /Date$/,
+    },
+  },
+};
+```
+
+Finally, update the `InboxScreen` stories and include a [parameter](https://storybook.js.org/docs/react/writing-stories/parameters) that mocks the remote API calls:
+
+```diff:title=src/components/InboxScreen.js
 import React from 'react';
-+ import { Provider } from 'react-redux';
-+ import { configureStore, createSlice } from '@reduxjs/toolkit';
-
-import { PureInboxScreen } from './InboxScreen';
-
-+ import * as TaskListStories from './TaskList.stories';
-
-+ // A super-simple mock of a redux store
-+  const Mockstore = configureStore({
-+    reducer: {
-+      tasks: createSlice({
-+        name: 'tasks',
-+        initialState: TaskListStories.Default.args.tasks,
-+        reducers: {
-+          updateTaskState: (state, action) => {
-+            const { id, newTaskState } = action.payload;
-+            const task = state.findIndex((task) => task.id === id);
-+            if (task >= 0) {
-+              state[task].state = newTaskState;
-+            }
-+          },
-+        },
-+      }).reducer,
-+    },
-+  });
+
+import InboxScreen from './InboxScreen';
+import store from '../lib/store';
++ import { rest } from 'msw';
+import { Provider } from 'react-redux';
 
 export default {
-  component: PureInboxScreen,
-+ decorators: [story => <Provider store={Mockstore}>{story()}</Provider>],
-  title: 'PureInboxScreen',
+  component: InboxScreen,
+  title: 'InboxScreen',
+  decorators: [(story) => <Provider store={store}>{story()}</Provider>],
 };
 
-const Template = args => <PureInboxScreen {...args} />;
+const Template = () => <InboxScreen />;
 
 export const Default = Template.bind({});
++ Default.parameters = {
++   msw: {
++     handlers: [
++       rest.get(
++         'https://jsonplaceholder.typicode.com/todos?userId=1',
++         (req, res, ctx) => {
++           return res(
++             ctx.json([
```

Comment:
Is this different than [`MockedState.tasks`](https://github.com/chromaui/learnstorybook.com/pull/579/files#diff-698e1d34eebfc7c303fa983175b1ef007fb84d6c583de7fe171ff69e0c7b22b5R188)? Should we export that so we can use it directly here, to more clearly communicate that this is the exact same data?

Reply:
Fixed with the latest commit. Thanks for the suggestion! Appreciate it. Marking it as solved